### DISCUSSION TITLE
BlockExecution

### DISCUSSION THREAD

**Discussion by 14jb02** [PSAppDeployToolkit/PSAppDeployToolkit#677] [General]
How to use BlockExecution without using Installation-Welcome?

I am trying to add the Wait-Process cmdlet to wait for a certain processes to stop before installing an application, and I wanted to use BlockExecution to prompt the user that there is application installing on the background when they attempting to launch the application.

I just want not to interrupt users with too many prompts to close this and that.

Can be it possible?

**Comment by mjr4077au** [comment]
The logic in Show-InstallationWelcome already handles everything you're doing. I'd test a default Deploy-Application script, with the CloseApps list amended to your needs and review how it works.
